wrappers/golang/runtime/memory.go

```diff
@@ -6,10 +6,6 @@ import "C"
 import "unsafe"
 
 func Malloc(size uint) (unsafe.Pointer, EIcicleError) {
-	if size == 0 {
-		return nil, AllocationFailed
-	}
-
```

Comment:
This together with the change made in `icicle_malloc` in cpp, its a breaking change.